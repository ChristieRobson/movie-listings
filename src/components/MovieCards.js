import React from 'react';
import { MovieCard } from './MovieCard';

export const MovieCards = (props) => {
    console.log('HERE', props);
    return (
        <div className="row">
            {props.movies.length > 0 &&
                props.movies.map((movie) =>
                    <MovieCard {...movie} key={movie.id}/>
                )
            }
        </div>
    )
};
