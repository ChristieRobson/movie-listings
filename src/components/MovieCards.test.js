import React from 'react';
import { shallow } from 'enzyme';
import { MovieCards } from './MovieCards';

it('renders MovieCards with 2 movies', () => {
    const moviesStub = [
        {
            id: 1,
            poster_path: 'abc123',
            title: 'A movie title',
            popularity: 88.123,
            genres : [
                {
                    "id": 28,
                    "name": "Action"
                },
                {
                    "id": 12,
                    "name": "Adventure"
                }
            ]
        }, {
            id: 2,
            poster_path: 'an456',
            title: 'Another movie ',
            popularity: 6.6,
            genres : [
                {
                    "id": 16,
                    "name": "Animation"
                }
            ]
        }
    ];
    let component = shallow(<MovieCards movies={moviesStub}/>);
    expect(component).toMatchSnapshot();
});