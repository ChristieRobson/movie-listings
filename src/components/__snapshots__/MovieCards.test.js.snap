// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieCards with 2 movies 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    movies={
        Array [
            Object {
              "genres": Array [
                Object {
                  "id": 28,
                  "name": "Action",
                },
                Object {
                  "id": 12,
                  "name": "Adventure",
                },
              ],
              "id": 1,
              "popularity": 88.123,
              "poster_path": "abc123",
              "title": "A movie title",
            },
            Object {
              "genres": Array [
                Object {
                  "id": 16,
                  "name": "Animation",
                },
              ],
              "id": 2,
              "popularity": 6.6,
              "poster_path": "an456",
              "title": "Another movie ",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          genres={
                    Array [
                              Object {
                                "id": 28,
                                "name": "Action",
                              },
                              Object {
                                "id": 12,
                                "name": "Adventure",
                              },
                            ]
          }
          id={1}
          popularity={88.123}
          poster_path="abc123"
          title="A movie title"
/>,
        <Unknown
          genres={
                    Array [
                              Object {
                                "id": 16,
                                "name": "Animation",
                              },
                            ]
          }
          id={2}
          popularity={6.6}
          poster_path="an456"
          title="Another movie "
/>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "genres": Array [
            Object {
              "id": 28,
              "name": "Action",
            },
            Object {
              "id": 12,
              "name": "Adventure",
            },
          ],
          "id": 1,
          "popularity": 88.123,
          "poster_path": "abc123",
          "title": "A movie title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "genres": Array [
            Object {
              "id": 16,
              "name": "Animation",
            },
          ],
          "id": 2,
          "popularity": 6.6,
          "poster_path": "an456",
          "title": "Another movie ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            genres={
                        Array [
                                    Object {
                                      "id": 28,
                                      "name": "Action",
                                    },
                                    Object {
                                      "id": 12,
                                      "name": "Adventure",
                                    },
                                  ]
            }
            id={1}
            popularity={88.123}
            poster_path="abc123"
            title="A movie title"
/>,
          <Unknown
            genres={
                        Array [
                                    Object {
                                      "id": 16,
                                      "name": "Animation",
                                    },
                                  ]
            }
            id={2}
            popularity={6.6}
            poster_path="an456"
            title="Another movie "
/>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "genres": Array [
              Object {
                "id": 28,
                "name": "Action",
              },
              Object {
                "id": 12,
                "name": "Adventure",
              },
            ],
            "id": 1,
            "popularity": 88.123,
            "poster_path": "abc123",
            "title": "A movie title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "genres": Array [
              Object {
                "id": 16,
                "name": "Animation",
              },
            ],
            "id": 2,
            "popularity": 6.6,
            "poster_path": "an456",
            "title": "Another movie ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
